build

```diff
@@ -89,11 +89,13 @@ COUNT=1
 
 for PKG_PATH in $(catkin_topological_order --only-folders || colcon list --topological-order --paths-only); do
   echo "::group::Building $COUNT/$TOTAL: $PKG_PATH"
-  test -f "$PKG_PATH/CATKIN_IGNORE" && echo "Skipped" && continue
-  test -f "$PKG_PATH/COLCON_IGNORE" && echo "Skipped" && continue
   (
   cd "$PKG_PATH"
-  GENERATED_DEBIAN_PACKAGE="ros-${ROS_DEB}$( (catkin_topological_order --only-names || colcon list --topological-order --names-only) | tr '_' '-')"
+  ROS_PACKAGE="$(catkin_topological_order --only-names || colcon list --topological-order --names-only)"
+  test -f "$PKG_PATH/CATKIN_IGNORE" && echo "$ROS_PACKAGE Skipped (CATKIN_IGNORE)" && exit
+  test -f "$PKG_PATH/COLCON_IGNORE" && echo "$ROS_PACKAGE Skipped (COLCON_IGNORE)" && exit
+  case " $SKIP_PACKAGES " in *" $ROS_PACKAGE "*) echo "$ROS_PACKAGE Skipped (in \$SKIP_PACKAGES)"; ;; esac
```

Comment:
If we have PACKAGE_A and PACKAGE_B, where PACKAGE_B depends on PACKAGE_A, will this work if we ignore PACKAGE_A? Will PACKAGE_B be skipped or will the build fail?

Reply:
It would fail. We can use `colcon --packages-skip-by-dep` to archive that once a newer Debian or Ubuntu is available as an Action image.

Reply:
I'm lost here, why would we need a new Debian or Ubuntu image? Can't we just use `colcon list --packages-skip-by-dep` now?

Reply:
`colcon` is only available in Debian bookworm and Ubuntu 22.10 and newer: https://repology.org/project/ros2-colcon-core/packages and I don't think it is worth it to do a backport. In principal we could use the packages from ros.org but I don't want to depend on third party stuff for the Action (the currently used PPA is only for the old Github VM as well) to make it easy to use.

Reply:
It's your project, but I don't see any problem with depending on the standard ROS tools from ros.org Alternatively, if you don't want to add a new APT sources line, we can always use colcon from PyPI and it'd work regardless of the distro

Reply:
And what about installing `colcon` from its GitHub repo via `pip`? Would that be enough for you?

But to be honest, I really don't see the problem with using the official ROS Debian packages, can you elaborate your reluctantness to do that? I'm trying to understand the rationale for that.

Reply:
This action should be simple to understand, verify and use by hand so people can understand how to build packages and trust it. Adding more sources, be it apt repos, PyPI or pip would make it harder to verify to be sure that no malicious code was sneaked in. I'm pretty sure that Github will provide newer VMs soon enough so we could have colcon at some point. For now I see listing all to be skipped packages as a valid option that should be fine in future as well.

Reply:
I see, I don't fully agree, but that's ok, thanks for the explanation. As a workaround we can just pass explicitly all the packages that depend on a given package.